fecgov/fec-cms
# feature/4189 cross origin links

## Summary

- Resolves #4189 

Lighthouse was warning us about links on our homepage that target `_blank`, that we should add `noopener` or `noreferrer` rules. Investigating the homepage, the only `target="_blank"` was the "upgrade your browser from IE" link to Browse Happy. So I added rules the `noreferrer` rule because do they really care/need to know that we may be linking some people to them?

Google's [definition](https://web.dev/external-anchors-use-rel-noopener/?utm_source=lighthouse&utm_medium=devtools):
> `rel="noopener"` prevents the new page from being able to access the window.opener property and ensures it runs in a separate process.
> `rel="noreferrer"` has the same effect but also prevents the Referrer header from being sent to the new page.

Then I started looking through the rest of the site.

This was my logic in making the changes I did:
- we don't need either for any links on our domain, so remove the rule for any links that are always internal.
- make it more clear that those are internal links (like changing `https://www.fec.gov/contact/` to `/contact/`)
- `noreferrer` as a starting point, being the most secure (Google, YouTube, 
- `noopener` when we may want to let others know that we're sending traffic to them (e.g., govdelivery)
- make a note (`NOREFERRER,NOOPENER?`) where I'm not sure which option may be best

**POSSIBLE ADDITIONAL CHANGES**
Is there any field in Wagtail that we should probably tighten a URL's `<a>` in the template?


### Required reviewers

Two?

## Impacted areas of the application

Potentially every off-site link. The only place I think it could remotely matter is for our GovDelivery, Google Calendar, or YouTube links.

## Screenshots

No visual changes

## Related PRs

None

## How to test

- pull the branch
- `./manage.py runserver`
- click some of these links to make sure they work
- Can run a Lighthouse scan on the homepage and any pages that used to have a `target="_blank"` but now includes `noopener` or `noreferrer`
- Should check that the Google Calendar links still work
- Are our GovDelivery or YouTube links breaking? (They could provide more information from us than this PR will let them see, but may be fine, too)
